packages/expo-doctor/src/doctor.ts

```diff
@@ -0,0 +1,103 @@
+import { getConfig } from '@expo/config';
+import chalk from 'chalk';
+
+import { ExpoConfigSchemaCheck } from './checks/ExpoConfigSchemaCheck';
+import { GlobalPrereqsVersionCheck } from './checks/GlobalPrereqsVersionCheck';
+import { IllegalPackageCheck } from './checks/IllegalPackageCheck';
+import { InstalledDependencyVersionCheck } from './checks/InstalledDependencyVersionCheck';
+import { SupportPackageVersionCheck } from './checks/SupportPackageVersionCheck';
+import { DoctorCheck, DoctorCheckParams } from './checks/checks.types';
+import { gteSdkVersion } from './utils/gteSdkVersion';
+import { logNewSection } from './utils/ora';
+import { warnUponCmdExe } from './warnings/windows';
+
+export async function runCheckAsync(
+  check: DoctorCheck,
+  checkParams: DoctorCheckParams
+): Promise<boolean> {
+  try {
+    const ora = logNewSection(check.description);
+    const result = await check.runAsync(checkParams);
+    if (!result.isSuccessful) {
```

Comment:
better to inverse this so the majority of code is one indentation lower, e.g. `if (result.isSuccessful) { return true; }`